### PR TITLE
feat: Collapsible Component

### PR DESCRIPTION
**Pull request by julianajlk**
Refactor of the Collapisble Component. It is now a composable component and set of subcomponents.

BREAKING CHANGE: The prop api has changed significantly to simplify usages and enable flexibility. See the docs for details.

BREAKING CHANGE: Related styles are included in paragon scss and no longer exist as a sibling to the component.

### CODE REVIEW

File: src/Collapsible/Collapsible.scss [hunk 1]
```diff
@@ -1,56 +1,43 @@
-// Local Variables
-$collapsible-border: 1px solid silver;
```
Comment (abutterworth) [inline src/Collapsible/Collapsible.scss]: Let's do our best to remove this file entirely and fold the core scss into /core and edx specific scss into edx/theme

File: src/icons/ChevronDown.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const ChevronDown = ({ height, ...other }) => (
+  <svg
+    height={height}
```
Comment (abutterworth) [inline src/icons/ChevronDown.jsx]: Any reason for height here?

Comment (julianajlk) [reply src/icons/ChevronDown.jsx]: Added height and width default to be 24px (1.5rem), the natural SVG size for these specific SVG icons

File: src/Collapsible/Collapsible.scss (hunk 1 above)
Comment (julianajlk) [reply src/Collapsible/Collapsible.scss]: To be done in the future

**Pull request merged**